src/tink/syntaxhub/FrontendContext.hx

```diff
@@ -100,21 +100,31 @@ class FrontendContext {
 	}
 	
 	static function moduleForType(name:String) {
-		if (name.indexOf('__impl') != -1 || plugins.getData().length == 0) return None;
+		if (name.indexOf('__impl') != -1 || plugins.getData().length == 0) return;
 		var pack = name.split('.');
-		var name = pack.pop();
-		var actual = pack.concat(['__impl', name]).join('.');
-		var exists = 
+		var tname = pack.pop();
+		var actual = pack.concat(['__impl', tname]).join('.');
+		cache[name] = {
+			pack: pack,
+			name: tname,
+			pos: Context.currentPos(),
+			fields: [],
+			kind: TDAlias(actual.asComplexType())
+		}
+		var exists =
 			try {
 				Context.getModule(actual);
 				true;
 			}
 			catch (e:Dynamic) false;
 		
 		if (!exists) {
-			var module = buildModule(pack, name);
-			if (module.types.length == 0) return None;
-			
+			var module = buildModule(pack, tname);
+			if (module.types.length == 0) {
+				cache.remove(name);
```

Comment:
Awesome, thanks, will look into this! :)

On Mon, May 23, 2016 at 7:04 PM, Jonas Malaco Filho <
notifications@github.com> wrote:

> In src/tink/syntaxhub/FrontendContext.hx
> <https://github.com/haxetink/tink_syntaxhub/pull/4#discussion_r64251970>:
>
> >  			try {
> >  				Context.getModule(actual);
> >  				true;
> >  			}
> >  			catch (e:Dynamic) false;
> >  		
> >  		if (!exists) {
> > -			var module = buildModule(pack, name);
> > -			if (module.types.length == 0) return None;
> > -			
> > +			var module = buildModule(pack, tname);
> > +			if (module.types.length == 0) {
> > +				cache.remove(name);
>
> This is the part that worries me the most.
>
> While this mimics the old behavior (cleanup of modules with no types), the
> eventual cache removal could mean that somewhere else a consumer of
> findType has now a reference to a module that no longer exists.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/haxetink/tink_syntaxhub/pull/4/files/bd4be093615819b1fc9905e5f455617f2fc999fe#r64251970>
>